client/src/redux/constants/credentialsConstants.js

```diff
@@ -1,9 +1,37 @@
-const LOAD_CREDENTIALS_ERROR = 'LOAD_CREDENTIALS_ERROR';
-const LOAD_CREDENTIALS_LOADING = 'LOAD_CREDENTIALS_LOADING';
-const LOAD_CREDENTIALS_SUCCESS = 'LOAD_CREDENTIALS_SUCCESS';
+const ADD_CREDENTIAL_ERROR = 'ADD_CREDENTIAL_ERROR';
+const ADD_CREDENTIAL_LOADING = 'ADD_CREDENTIAL_LOADING';
+const ADD_CREDENTIAL_SUCCESS = 'ADD_CREDENTIAL_SUCCESS';
+
+const DELETE_CREDENTIAL_ERROR = 'DELETE_CREDENTIAL_ERROR';
+const DELETE_CREDENTIAL_LOADING = 'DELETE_CREDENTIAL_LOADING';
+const DELETE_CREDENTIAL_SUCCESS = 'DELETE_CREDENTIAL_SUCCESS';
+
+const GET_CREDENTIAL_ERROR = 'GET_CREDENTIAL_ERROR';
+const GET_CREDENTIAL_LOADING = 'GET_CREDENTIAL_LOADING';
+const GET_CREDENTIAL_SUCCESS = 'GET_CREDENTIAL_SUCCESS';
+
+const GET_CREDENTIALS_ERROR = 'GET_CREDENTIALS_ERROR';
+const GET_CREDENTIALS_LOADING = 'GET_CREDENTIALS_LOADING';
+const GET_CREDENTIALS_SUCCESS = 'GET_CREDENTIALS_SUCCESS';
+
+const UPDATE_CREDENTIAL_ERROR = 'UPDATE_CREDENTIAL_ERROR';
+const UPDATE_CREDENTIAL_LOADING = 'UPDATE_CREDENTIAL_LOADING';
+const UPDATE_CREDENTIAL_SUCCESS = 'UPDATE_CREDENTIAL_SUCCESS';
```

Comment:
The naming here is a bit confusing. the `ADD` constants seem like I'm adding something rather than updating the ADD call results.

I'm wondering if the actions should simplify and just have CREDENTIAL_ERROR, CREDENTIALS_LOADING, CREDENTIALS_SUCCESS

Then the action can have more info in it as to which call it was for? Some of these constant defs get to be a rather long list (like scans 😮 )

You can leave as is for now and we can address later.

Reply:
Indeed they do get long in name and the sheer number of them. 

I'm possibly up for trying to handle more in the actions themselves which, agreed, for sure would need an additional check if they're not broken out here. Though either way somethings probably going to get lengthy, if it's an additional switch/label we pass to, or more likely a value we pass back to the subscriber, that may work... like ```actionType: 'error'``` and we simply reduce the constants to... though this could also be in "addition too" 

```js
// within a component
import { credentialTypes } from '../../redux/constants';
...
Store.dispatch({ type: credentialTypes.ADD});
...
```

As a start in a follow up PR can look at something as simple as removing "CREDENTIAL" from the name itself and making sure to use the namespacing when the constants are imported... like.

```js
// within an Action...
import { credentialTypes } from '../constants';
...
Store.dispatch({ type: credentialTypes.ADD_ERROR });
...
```

The nice part is, it's already lengthy, deleting/removing things feels pretty satisfying when it comes around =)